src/prefect/tasks/great_expectations/checkpoints.py

```diff
@@ -10,108 +10,249 @@
+                this tasks. Defaults to `False`.
 
         Raises:
             - 'signals.VALIDATIONFAIL' if the validation was not a success
+
         Returns:
             - result
                 ('great_expectations.validation_operators.types.validation_operator_result.ValidationOperatorResult'):
                 The Great Expectations metadata returned from the validation
 
         """
-
         if checkpoint_name is None:
             raise ValueError("You must provide the checkpoint name.")
 
         runtime_environment = runtime_environment or dict()
 
-        context = ge.DataContext(
-            context_root_dir=context_root_dir, runtime_environment=runtime_environment
-        )
-        checkpoint = context.get_checkpoint(checkpoint_name)
+        # Load context if not provided directly
+        if not context:
+            context = ge.DataContext(
+                context_root_dir=context_root_dir,
+                runtime_environment=runtime_environment,
+            )
+
+        # if assets are not provided directly through `assets_to_validate` then they need be loaded
+        #   if the checkpoint is being loaded from the context then load suite and batch from there
+        #   otherwise get batch from `batch_kwargs` and `expectation_suite_name`
+        if not assets_to_validate:
+            assets_to_validate = []
+            if get_checkpoint_from_context:
+                ge_checkpoint = context.get_checkpoint(checkpoint_name)
 
-        batches_to_validate = []
-        for batch in checkpoint["batches"]:
-            batch_kwargs = batch["batch_kwargs"]
-            for suite_name in batch["expectation_suite_names"]:
-                suite = context.get_expectation_suite(suite_name)
-                batch = context.get_batch(batch_kwargs, suite)
-                batches_to_validate.append(batch)
+                for batch in ge_checkpoint["batches"]:
+                    batch_kwargs = batch["batch_kwargs"]
+                    for suite_name in batch["expectation_suite_names"]:
+                        suite = context.get_expectation_suite(suite_name)
+                        batch = context.get_batch(batch_kwargs, suite)
+                        assets_to_validate.append(batch)
+            else:
+                assets_to_validate.append(
+                    context.get_batch(batch_kwargs, expectation_suite_name)
+                )
 
+        # Run validation operator
         results = context.run_validation_operator(
-            checkpoint["validation_operator_name"],
-            assets_to_validate=batches_to_validate,
-            run_id={"run_name": prefect.context.get("task_slug")},
+            checkpoint_name or ge_checkpoint["validation_operator_name"],
+            assets_to_validate=assets_to_validate,
+            run_id={"run_name": run_name or prefect.context.get("task_slug")},
         )
 
         if results.success is False:
             raise signals.VALIDATIONFAIL(result=results)
```

Comment:
This is the only place we use this signal / state; I do remember the original motivation but now I'm inclined to think we should enter a standard `Failed` state instead